src/app/modules/payments/payments.service.ts

```diff
@@ -80,3 +80,43 @@ export const findPaymentBySubmissionId = (
     return okAsync(payment)
   })
 }
+
+export const updateReceiptUrlAndTransactionFee = (
+  paymentId: IPaymentSchema['_id'],
+  receiptUrl: string,
+  stripeTransactionFee: number,
+): ResultAsync<IPaymentSchema, PaymentNotFoundError | DatabaseError> => {
+  // Retrieve payment object from database and
+  // Update payment's receipt url
+  return ResultAsync.fromPromise(
+    PaymentModel.findByIdAndUpdate(
+      paymentId,
+      {
+        $set: {
+          receiptUrl,
+          stripeTransactionFee,
+          updatedAt: new Date(),
```

Comment:
do we need to explicitly update this field? I expected that it would be updated automatically, as defined by the `timestamps` in the model

Reply:
[073bdd5](https://github.com/opengovsg/FormSG/pull/5942/commits/073bdd55fcd76bf9fd8d4008cec21e733fee47d4)